### PR TITLE
NativeAOT build support and doc improvements

### PR DESCRIPTION
**Pull request by jkotas**
- Add IlcInstructionSet property that can be used to set the target instruction set from the command line.
- Moved link to the documentation of reflection-free mode to be in less prime location, so that people trying NativeAOT for the first time are not steared towards it.
- Copied help text improvements from crossgen2

### CODE REVIEW

**Comment by EgorBo** [comment]
It'd be nice to be able to do `--instruction-set:host` or `--instruction-set:native` similar to `-mcpu=native` in C++
I noticed several times cases where developers specified e.g. `--instruction-set:avx2` but forgot about various useful BMI1-2, LZCNT instructions which aren't implied by `avx2`

**Review by MichalStrehovsky — APPROVED** [review]
I haven't exposed the instruction set argument because I think we might need to preserve some extra registers in transitions for AVX that we currently don't. I don't have much experience in that area so I'll leave that up to you. Thanks!

**Comment by jkotas** [comment]
> I think we might need to preserve some extra registers in transitions for AVX that we currently don't.

We will need to deal with AVX registers for https://github.com/dotnet/runtime/issues/67805. We are fine as of today.

File: src/coreclr/nativeaot/docs/reflection-free-mode.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ Reflection-free mode is a mode of the NativeAOT compiler and runtime that greatl
 * Reduced working set and better code locality - parts of the program are more tightly packed together.
 * Less metadata for people to reverese engineer - apps compiled in reflection-free mode are as hard to reverse engineer as apps written in e.g. C++.
```
Comment (danmoseley) [inline src/coreclr/nativeaot/docs/reflection-free-mode.md:7]: Typo..reverse

File: src/coreclr/nativeaot/docs/reflection-in-aot-mode.md [hunk 2]
```diff
@@ -93,3 +93,7 @@ To enable simulated `Assembly.GetCallingAssembly`, you will need:
   <ItemGroup>
     <RuntimeHostConfigurationOption Include="Switch.System.Reflection.Assembly.SimulatedCallingAssembly" Value="true" />
   </ItemGroup>
+
+## Experimental Reflection Free Mode
+
+Reflection-free mode is a an experimental mode of the NativeAOT compiler and runtime that greatly reduces the functionality of the reflection APIs and demonstrates how far can reflection trimming can get. See [Reflection Free Mode](reflection-free-mode.md) for mode details.
```
Comment (danmoseley) [inline src/coreclr/nativeaot/docs/reflection-in-aot-mode.md:99]: "How far reflection can"

File: src/coreclr/nativeaot/docs/optimizing.md [hunk 3]
```diff
@@ -36,15 +36,14 @@ To aid in troubleshooting some of the most common problems related to trimming a
...
-* `<IlcFoldIdenticalMethodBodies>true</IlcFoldIdenticalMethodBodies>`: folds method bodies with identical bytes (method body deduplication). This makes your app smaller, but the stack traces might sometimes look nonsensical (unexpected methods might show up in the stack trace because the expected method had the same bytes as the unexpected method). Note: the current implementation of deduplication doesn't attempt to make the folding unobservable to managed code: delegates pointing to two logically different methods that ended up being folded together will compare equal.
+* `<IlcInstructionSet>`: By default, the compiler targets the minimum instruction set supported by the target OS and architecture. This option allows targeting newer instruction sets for better performance. The native binary will require the instruction sets to be supported by the hardware in order to run. For example, `<IlcInstructionSet>avx2,bmi2,fma,pclmul,popcnt,aes</IlcInstructionSet>` will produce binary that takes advantage of instruction sets that are typically present on current Intel and AMD processors. Run `ilc.exe --help` for the full list of available instruction sets.
```
Comment (adamsitnik) [inline src/coreclr/nativeaot/docs/optimizing.md:43]: Please excuse me if this is a dumb question, but where can I find `ilc.exe`?

Comment (jkotas) [reply src/coreclr/nativeaot/docs/optimizing.md:43]: Does this help? https://github.com/dotnet/runtime/pull/68117

**Pull request merged**